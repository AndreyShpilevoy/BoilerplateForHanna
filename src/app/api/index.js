import getUserProfilesArrayApi from 'api/__fakeApi__/getUserProfilesArrayApi';
import getDivisionsArrayApi from 'api/__fakeApi__/getDivisionsArrayApi';
import getLocaleApi from 'api/__fakeApi__/getLocaleApi';

export {
    getUserProfilesArrayApi,
    getDivisionsArrayApi,
    getLocaleApi
};
