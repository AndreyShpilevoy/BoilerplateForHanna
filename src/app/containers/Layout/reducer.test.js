/* eslint no-undef: 0, max-statements:0, import/first: 0 */

jest.mock('api');
import {getLocaleApi} from 'api';
import {
    getLocale,
    getLocaleSuccess,
    getLocaleError,
    GET_LOCALE,
    GET_LOCALE_SUCCESS,
    GET_LOCALE_FAILED,
    layoutReducer,
    getLocaleNonBlockSaga,
    getLocaleSaga,
    layoutSaga
} from './reducer';

describe('Conference reducer', () => {
    it('getLocale should create expected object', () => {
        const expectedResult = {type: GET_LOCALE};
        expect(getLocale()).toEqual(expectedResult);
    });

    it('getLocaleSuccess should create expected object', () => {
        const expectedResult = {
            type: GET_LOCALE_SUCCESS,
            payload: {
                locale: 'NL'
            }
        };
        expect(getLocaleSuccess('NL')).toEqual(expectedResult);
    });

    it('getLocaleError should create expected object', () => {
        const expectedResult = {
            type: GET_LOCALE_FAILED,
            payload: {error: {text: 'error'} }
        };
        expect(getLocaleError({text: 'error'})).toEqual(expectedResult);
    });

    it('layoutReducer with action GET_LOCALE_SUCCESS should return expected state', () => {
        const defaulState = {
            locale: 'EN',
            error: {}
        };
        const action = {
            type: GET_LOCALE_SUCCESS,
            payload: {
                locale: 'NL'
            }
        };
        const expectedResult = {
            locale: 'NL',
            error: {}
        };
        expect(layoutReducer(defaulState, action)).toEqual(expectedResult);
    });

    it('layoutReducer with action GET_LOCALE_FAILED should return expected state', () => {
        const defaulState = {
            locale: 'EN',
            error: {}
        };
        const action = {
            type: GET_LOCALE_FAILED,
            payload: {
                error: {text: 'error'}
            }
        };
        const expectedResult = {
            locale: 'EN',
            error: {text: 'error'}
        };
        expect(layoutReducer(defaulState, action)).toEqual(expectedResult);
    });

    it('layoutReducer with invald (ACTION) action should return expected state', () => {
        const defaulState = {
            locale: 'EN',
            error: {}
        };
        const action = {
            type: 'ACTION',
            payload: 'something'
        };
        const expectedResult = {
            locale: 'EN',
            error: {}
        };
        expect(layoutReducer(defaulState, action)).toEqual(expectedResult);
    });

    it('getLocaleSaga should be in loop and return expected values', () => {
        const generator = getLocaleSaga();

        const firstYield = generator.next();
        const secondYield = generator.next();
        const thirdYield = generator.next();

        expect(firstYield).toMatchSnapshot();
        expect(secondYield).toMatchSnapshot();
        expect(secondYield.value.FORK.fn.name).toMatchSnapshot();
        expect(thirdYield).toMatchSnapshot();
    });

    it('getLocaleNonBlockSaga on success should return 2 expected yields. 3 yield should be in state Done', () => {
        const generator = getLocaleNonBlockSaga();
        const locale = 'RU';

        const firstYield = generator.next();
        const secondYield = generator.next({locale});
        const thirdYield = generator.next();

        expect(firstYield).toMatchSnapshot();
        expect(firstYield.value.CALL.fn.name).toMatchSnapshot();
        expect(secondYield).toMatchSnapshot();
        expect(thirdYield).toMatchSnapshot();
    });

    it('getLocaleNonBlockSaga on fail should return 2 expected yields. 3 yield should be in state Done', () => {
        const generator = getLocaleNonBlockSaga();
        const error = 'some error';

        const firstYield = generator.next();
        const secondYield = generator.next({error});
        const thirdYield = generator.next();

        expect(firstYield).toMatchSnapshot();
        expect(firstYield.value.CALL.fn.name).toMatchSnapshot();
        expect(secondYield).toMatchSnapshot();
        expect(thirdYield).toMatchSnapshot();
    });

    it('layoutSaga should return 1 yield with 1 saga. 2d yield should be in state Done', () => {
        const generator = layoutSaga();
        expect(generator.next()).toMatchSnapshot();
    });
});