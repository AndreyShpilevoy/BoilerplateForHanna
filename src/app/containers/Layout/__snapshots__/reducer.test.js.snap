// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Conference reducer getLocaleNonBlockSaga on fail should return 2 expected yields. 3 yield should be in state Done 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`Conference reducer getLocaleNonBlockSaga on fail should return 2 expected yields. 3 yield should be in state Done 2`] = `"getLocaleApi"`;

exports[`Conference reducer getLocaleNonBlockSaga on fail should return 2 expected yields. 3 yield should be in state Done 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "payload": Object {
          "error": "some error",
        },
        "type": "GET_LOCALE_FAILED",
      },
      "channel": null,
    },
  },
}
`;

exports[`Conference reducer getLocaleNonBlockSaga on fail should return 2 expected yields. 3 yield should be in state Done 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Conference reducer getLocaleNonBlockSaga on success should return 2 expected yields. 3 yield should be in state Done 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "CALL": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`Conference reducer getLocaleNonBlockSaga on success should return 2 expected yields. 3 yield should be in state Done 2`] = `"getLocaleApi"`;

exports[`Conference reducer getLocaleNonBlockSaga on success should return 2 expected yields. 3 yield should be in state Done 3`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "PUT": Object {
      "action": Object {
        "payload": Object {
          "locale": "RU",
        },
        "type": "GET_LOCALE_SUCCESS",
      },
      "channel": null,
    },
  },
}
`;

exports[`Conference reducer getLocaleNonBlockSaga on success should return 2 expected yields. 3 yield should be in state Done 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`Conference reducer getLocaleSaga should be in loop and return expected values 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "TAKE": Object {
      "pattern": "GET_LOCALE",
    },
  },
}
`;

exports[`Conference reducer getLocaleSaga should be in loop and return expected values 2`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "FORK": Object {
      "args": Array [],
      "context": null,
      "fn": [Function],
    },
  },
}
`;

exports[`Conference reducer getLocaleSaga should be in loop and return expected values 3`] = `"getLocaleNonBlockSaga"`;

exports[`Conference reducer getLocaleSaga should be in loop and return expected values 4`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "TAKE": Object {
      "pattern": "GET_LOCALE",
    },
  },
}
`;

exports[`Conference reducer layoutSaga should return 1 yield with 1 saga. 2d yield should be in state Done 1`] = `
Object {
  "done": false,
  "value": Object {
    "@@redux-saga/IO": true,
    "ALL": Array [
      GeneratorFunctionPrototype {
        "_invoke": [Function],
      },
    ],
  },
}
`;
